pkg/apis/core/v1alpha1/storagecluster.go

```diff
@@ -118,18 +118,21 @@ type SecuritySpec struct {
 // AuthSpec lets the user define authorization configurations
 // for creating a PX Security enabled cluster
 type AuthSpec struct {
-	Authenticators *AuthenticatorsSpec `json:"authenticators,omitempty"`
-}
-
-// AuthenticatorsSpec defines which auth mechanisms are setup for PX Security
-type AuthenticatorsSpec struct {
 	SelfSigned *SelfSignedSpec `json:"selfSigned,omitempty"`
 }
 
 // SelfSignedSpec defines a configuration for self signed authentication
 type SelfSignedSpec struct {
-	Issuer        *string        `json:"issuer,omitempty"`
-	TokenLifetime *meta.Duration `json:"tokenLifetime,omitempty"`
+	Issuer        *string             `json:"issuer,omitempty"`
+	TokenLifetime *meta.Duration      `json:"tokenLifetime,omitempty"`
+	SharedSecret  *SecretKeyReference `json:"sharedSecretRef,omitempty"`
+}
+
+// SecretKeyReference is a user-friendly reference to a value based on a secret name and key
+type SecretKeyReference struct {
```

Comment:
I made this new object instead of using the k8s built in ones. I think having a custom one with secretName/secretNamespace would be better.

Reply:
I think we should only take the secret name, just like we do for image pull secret and kvdb secret. Kubernetes does the same for image pull secrets in pods. Just trying to keep the spec as simple as possible.
We can mandate the secret to be present in the StorageCluster's namespace. 
For the key, we can assume a constant which the operator knows and use the same in our docs. If we don't find the correct key in the secret we can raise an event, that way users know what is the expected key in-case they created the secret incorrectly.
For most cases the operator would be creating the secret, so the key should be the one expected.

Reply:
Ah yeah good point. That simplifies things a lot.